libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routefetch/RoutesViewModel.kt

```diff
@@ -49,7 +80,53 @@ internal class RoutesViewModel : UIViewModel<Unit, RoutesAction>(Unit) {
         }
     }
 
-    private fun fetchRoute(options: RouteOptions, mapboxNavigation: MapboxNavigation) {
+    @Suppress("MissingPermission")
+    private fun startNavigation(mapboxNavigation: MapboxNavigation) {
+        with(mapboxNavigation) {
+            // Temporarily trigger replay here
+            mapboxReplayer.clearEvents()
+            resetTripSession()
+            mapboxReplayer.pushRealLocation(navigationOptions.applicationContext, 0.0)
+            mapboxReplayer.play()
+            startReplayTripSession()
+        }
+    }
+
+    private fun stopNavigation(mapboxNavigation: MapboxNavigation) {
+        with(mapboxNavigation) {
+            // Stop replay here
+            mapboxReplayer.clearEvents()
+            resetTripSession()
+        }
+    }
+
+    private fun shouldReloadRoute(
+        oldDestination: Destination?,
+        newDestination: Destination?
+    ) = oldDestination != newDestination &&
+        newDestination != null &&
+        navigationState.value == NavigationState.RoutePreview
+
+    private fun fetchAndSetRoute(
+        mapboxNavigation: MapboxNavigation,
+        destination: Destination?,
+        cb: (() -> Unit)? = null
+    ) {
+        val from = locationState.value?.toPoint()
```

Comment:
This looks like a valid case to depend on `LocationViewModel` for now. Replay the locationState flow with the state flow in `LocationViewModel`